docs/examples/compute/instance/variables.tf

```diff
@@ -8,6 +8,14 @@ variable "compartment_ocid" {}
 variable "ssh_public_key" {}
 variable "ssh_private_key" {}
 
+variable "NumInstances" {
```

Comment:
If I deploy with one set of values for these 2 variables and then change them, will it only create/destroy what is required, or will it destroy and recreate other resources?

Reply:
Good question. The answer is it depends. If you change the NumInstances, then only the required resources will be created/destroyed.

If you modify the NumVolumesPerInstance, this could result in some volumes being reattached to other instances. I'm trying to think of a way to solve this. The interpolation syntax doesn't really have a way to track the previous mappings between volumes/instances.

Reply:
If you can solve it, great. It wouldn't surprise me if there isn't a good solution for this though, in which case maybe a comment would be good.

Reply:
Yep. I will add a comment for this.

I couldn't find a good way to preserve the mappings with interpolation. Otherwise, you would need to manually duplicate the volumes and attachments to avoid using interpolation.

Reply:
Addressed in latest update.